loopchain/blockchain/blockchain.py

```diff
@@ -269,25 +272,21 @@ def add_block(self,
 
     def __add_block(self, block: Block, confirm_info, need_to_write_tx_info=True, need_to_score_invoke=True):
         with self.__add_block_lock:
-            invoke_results = self.__invoke_results.get(block.header.hash.hex(), None)
-            if invoke_results is None and need_to_score_invoke:
-                if block.header.height == 0:
-                    block, invoke_results = ObjectManager().channel_service.genesis_invoke(block)
-                else:
-                    block, invoke_results = ObjectManager().channel_service.score_invoke(block, self.__last_block)
+            receipts, next_prep = self.__invoke_results.get(block.header.hash.hex(), (None, None))
+            if receipts is None and need_to_score_invoke:
+                self.get_invoke_func(block.header.height)(block, self.__last_block)
```

Comment:
Are you sure block.prev_hash ==  self.__last_block.hash?
What if the condition won't be meet?

Reply:
I think, It's a same logic as before. There may be a problem but this is not a concern in this PR.